zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs

```diff
@@ -201,21 +226,16 @@ pub(crate) async fn get_gateway_migration_state(
         // No event has been sent.
         // It means that either migration has not yet started or
         // the chain admin has completed the migration but without sending any notiifcation.
+        // Or the latest migration was long time ago.
 
         // Firslty check for consistency with the server.
-        if gateway_migration_status.latest_notification.is_some() {
-            anyhow::bail!(format!(
-                "Server has seen an event not present on L1. Status {:#?}",
-                gateway_migration_status
-            ));
+        if gateway_migration_status.state == GatewayMigrationState::InProgress {
+            anyhow::bail!("Very old migration in progress");
         }
 
         let Some(current_sl_from_server) = gateway_migration_status.settlement_layer else {
-            todo!(
-                "
-                The migration has started, but the settlement layer is not defined. \
-                Waiting manual finalization or L2 L1 transaction"
-            );
+            // It means that the server is in the middle of some migration.
+            anyhow::bail!("Very old migration in progress");
```

Comment:
Using direction, you can return that it requires manual finalization. or waiting for gateway to process the tx

Reply:
This entire clause is for the case when I could not find L1 migration event on L1, so the event is either very old or there has been no migration before. 

If the event is too old, we dont handle such case yet